collectors/mesos/agent/agent_test.go

```diff
@@ -349,18 +349,27 @@ func TestTransform(t *testing.T) {
 		if err := json.Unmarshal(mockAgentState, &mac.agentState); err != nil {
 			panic(err)
 		}
-		if err := json.Unmarshal(deficientContainerMetrics, &mac.containerMetrics); err != nil {
-			panic(err)
-		}
-
-		Convey("Should return a []producers.MetricsMessage without errors", func() {
+		Convey("With container metrics", func() {
+			if err := json.Unmarshal(mockContainerMetrics, &mac.containerMetrics); err != nil {
```

Comment:
Yep, I just tested this by adding to a counter each time it was re-initialized and asserting in each sub `Convey` method that it had incremented.